TestHelpers/struct_field_to_index_map_generator.py

```diff
@@ -0,0 +1,312 @@
+#! python3 struct_field_to_index_map_generator.py <ast txt file> <output for map file>
+# generate AST text with: clang -Xclang -ast-dump -fsyntax-only -fno-color-diagnostics <c file> > <ast txt file>
+
+# example (run in build dir):
+# clang -Xclang -ast-dump -fsyntax-only -fno-color-diagnostics ../test/simple_ptr_test/index.c > ast.txt ; python3 ../TestHelpers/
+# struct_field_to_index_map_generator.py ast.txt map.txt
+
+# finds every variable whose type is a struct and creates a map of it and its field names to it and its field indices.
+# example:
+'''
+struct my_struct {
+ char* x;
+ char* y;
+}
+
+int main() {
+    struct my_struct A;
+    struct my_struct B;
+}
+'''
+# this C code will generate the following map:
+'''
+[A.x] => [A.0]
+[A.y] => [A.1]
+[B.x] => [B.0]
+[B.y] => [B.1]
+'''
+# this is used for Testers/Passes when we want to specify
+# a field of a variable by field name (rather than index)
+# to test its MustCall/CalledMethods
+
+import sys
+
+DEBUG = True
+
+
+def logout(x: str):
+    if (DEBUG):
+        print(x)
+
+
+class Field:
+    def __init__(self, field_name: str, field_index: int):
+        self.field_name = field_name
+        self.field_index = field_index
+
+
+class Struct:
+    def __init__(self, name: str):
+        self.name = name
+        self.typedefs: list[str] = []
+        self.fields: list[Field] = []
+
+    def get_name(self) -> str:
+        return self.name
+
+    def get_typedefs(self) -> "list[str]":
+        return self.typedefs
```

Comment:
No `get_fields`? That's surprising.

Reply:
That was because I was defining setters and getters but they weren't actually being enforced so I could still just get `.fields`; fixing by making the members of `Struct` private by adding `__` to their name (e.g., `self.name` => `self.__name`)